model-engine/tests/unit/inference/test_vllm_batch.py

```diff
@@ -0,0 +1,308 @@
+import json
+from unittest.mock import MagicMock, call, mock_open, patch
+
+import pytest
+from model_engine_server.common.dtos.llms import CompletionOutput, TokenOutput
+from model_engine_server.inference.batch_inference.vllm_batch import batch_inference
+
+
+@pytest.mark.asyncio
+@patch("model_engine_server.inference.batch_inference.vllm_batch.CreateBatchCompletionsRequest")
+@patch(
+    "model_engine_server.inference.batch_inference.vllm_batch.CreateBatchCompletionsRequestContent"
+)
+@patch("model_engine_server.inference.batch_inference.vllm_batch.generate_with_vllm")
+@patch("model_engine_server.inference.batch_inference.vllm_batch.get_s3_client")
+@patch("subprocess.Popen")
+@patch(
+    "model_engine_server.inference.batch_inference.vllm_batch.smart_open.open",
+    new_callable=mock_open,
+    read_data="Mocked content",
+)
+async def test_batch_inference(
+    mock_open_func,
+    mock_popen,
+    mock_get_s3_client,
+    mock_generate_with_vllm,
+    mock_create_batch_completions_request_content,
+    mock_create_batch_completions_request,
+    create_batch_completions_request,
+    create_batch_completions_request_content,
+    create_vllm_request_outputs,
+    mock_s3_client,
+    mock_process,
+):
+    # Mock the necessary objects and data
+    mock_popen.return_value = mock_process
```

Comment:
nit: you can create a new fixture for that mocks the return value for subprocess.Popen using the `mock_process` fixture such that you don't need to make this declaration at the start of each unit test. it'll clean up logic here:
```
def test(mock1, mock2):
  mock2.return_value = mock1
```
becomes
```
def test(mock2):
  ...
```

you can do that for several of the mocks done at the start of each unit test